crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -120,6 +120,7 @@ impl<'db> ScopeId<'db> {
             NodeWithScopeKind::ClassTypeParameters(_)
                 | NodeWithScopeKind::FunctionTypeParameters(_)
                 | NodeWithScopeKind::Function(_)
+                | NodeWithScopeKind::TypeAlias(_)
```

Comment:
On closer look here I realize that `NodeWithScopeKind::TypeAliasTypeParameters` also belongs here, just like `ClassTypeParameters` and `FunctionTypeParameters`.

Though it seems like we could also simplify this to match on `self.node(db).kind()` instead, and then including `ScopeKind::Annotation` would cover all three of the type-param scopes.

Either option seems fine to me.

Reply:
> Though it seems like we could also simplify this to match on `self.node(db).kind()` instead, and then including `ScopeKind::Annotation` would cover all three of the type-param scopes.

If I understand correctly, you would want to change it to:
```rs
matches!(
    self.node(db).scope_kind(),
    ScopeKind::Annotation | ScopeKind::Function | ScopeKind::TypeAlias | ScopeKind::Comprehension
)
```
This would match everything that was matched previously, but it would additionally iunclude `NodeWithScopeKind::Lambda(_)`. I wasn't sure if that's correct, so I just added `NodeWithScopeKind::TypeAliasTypeParameters` for now.